include/envoy/network/dns.h

```diff
@@ -35,10 +47,10 @@ class DnsResolver {
 
   /**
    * Called when a resolution attempt is complete.
-   * @param address_list supplies the list of resolved IP addresses. The list will be empty if
+   * @param response supplies the list of resolved IP addresses and TTLs. The list will be empty if
    *                     the resolution failed.
    */
-  using ResolveCb = std::function<void(std::list<Address::InstanceConstSharedPtr>&& address_list)>;
+  using ResolveCb = std::function<void(const std::list<DnsResponse>& response)>;
```

Comment:
I don't feel strongly about it but why not still provide the list via move semantics and let the caller move it if they want?

Reply:
It's hard to tell which one is better. If you don't insist, I would keep this change.

Reply:
OK, sorry, but I will insist. :) I don't see any reason not to offer move semantics for a list which is a temporary anyway.

/wait

Reply:
I am ok to revert the change.

Reply:
IIRC, at the first version it was `const list &&`. 
During the waiting somehow it changed to `list &&` and it is proved correct. And `list &&` beats `const list &`